drivers/dma/dma_handlers.c

```diff
@@ -24,3 +24,39 @@ static inline int z_vrfy_dma_stop(const struct device *dev, uint32_t channel)
 	return z_impl_dma_stop((const struct device *)dev, channel);
 }
 #include <zephyr/syscalls/dma_stop_mrsh.c>
+
+static int z_vrfy_dma_chan_filter(const struct device * dev, int channel, void * filter_param)
+{
+	K_OOPS(K_SYSCALL_DRIVER_DMA(dev, chan_filter));
+	return z_impl_dma_chan_filter((const struct device *)dev, channel, filter_param);
```

Comment:
@teburd  I marked this as draft for now, but wanted to share for some early comments.

I'm not 100% sure if this is enough to verify given filter param is a pointer. Most DMA drivers I checked just use the 'filter_param' to pass a value, so this would also work from user-space. But of course, if an actual pointer to something is passed, that won't work from user-space....